### ISSUE TITLE
React-Slick does not work on SSR projects

### ISSUE THREAD

**Issue by ShreyKumar** [akiran/react-slick#1837]
I am trying to install react-slick on a NextJS project with [this example](https://codesandbox.io/s/react-slick-playground-8d6h5) but none of the images are showing up.

[As seen here](https://drive.google.com/file/d/1inXscYdXx5Hur6-jA7fSjNnqI5rALqIg/view?usp=sharing), the blue rectangle on the left is the slick `Slider` component which does not display any image at all.

**Comment by abhishekashyap** [comment]
You also need to import slick-carousel css

Install it using:
```bash
npm install slick-carousel
```

and import it in **_app.js** using :

```js
import "slick-carousel/slick/slick.css";
```

**Comment by MikeMajara** [comment]
@katsos, can you explain what not worked?

@abhishekashyap it's probably worth mentioning that the style only needs to be imported wherever the Slider is used.

**Comment by katsos** [comment]
You have to use https://www.npmjs.com/package/@ant-design/react-slick to work with SSR without problems.

I was having problems with ssr-client prop matching
```
react-dom.development.js?61bb:88 Warning: Prop `src` did not match. Server: "/assets/img/categories/packaged.png" Client: "/assets/img/categories/baby.png"
``` 

Sample of source code
```

const CATEGORIES_MAP = {
  '01': {
    imgName: 'vegetables',
  },
  '02': {
    imgName: 'meat',
  },
  '03': {
    imgName: 'milk',
  },
  '05': {
    imgName: 'sausage',
  },
  '04': {
    imgName: 'cheese',
  },
  '06': {
    imgName: 'frozen',
  },
  '07': {
    imgName: 'packaged',
  },
  '08': {
    imgName: 'alcohol',
  },
  '09': {
    imgName: 'pet',
  },
  '10': {
    imgName: 'personal',
  },
  '11': {
    imgName: 'house-cleaning',
  },
  '12': {
    imgName: 'house',
  },
  '13': {
    imgName: 'baby',
  },
  '14': {
    imgName: 'fridge',
  },
  '15': {
    imgName: 'bio',
  },
  '17': {
    imgName: 'heaters',
  },
};

const SLIDER_SETTINGS = {
  adaptiveHeight: true,
  className: styles.slider,
  infinite: true,
  slidesToShow: 8,
  slidesToScroll: 8,
  speed: 500,
  prevArrow: <PrevArrow category callback={() => arrowClick('categoryLeftClick')} />,
  nextArrow: <NextArrow category callback={() => arrowClick('categoryRightClick')} />,
  responsive: [
    {
      breakpoint: 1200,
      settings: {
        slidesToShow: 7,
        slidesToScroll: 7,
      },
    },
    {
      breakpoint: 1120,
      settings: {
        slidesToShow: 6,
        slidesToScroll: 6,
      },
    },
    {
      breakpoint: 1024,
      settings: {
        slidesToShow: 5,
        slidesToScroll: 5,
      },
    },
    {
      breakpoint: 824,
      settings: {
        slidesToShow: 3,
        slidesToScroll: 3,
      },
    },
    {
      breakpoint: 578,
      settings: {
        slidesToShow: 1,
        slidesToScroll: 1,
      },
    },
  ],
};

export default function Categories({categories}) {
  return (
    <section className={styles.container}>
      <Slider {...SLIDER_SETTINGS}>
        {categories.map(({_id, divisionCode, name, numberOfProductsToDisplay, slug}) => (
          <div key={`${_id}${numberOfProductsToDisplay}`}>
            <Link href="/products/[...slug]" as={`/products/${slug}`}>
              <a className={styles.tile}>
                <div className={styles.imgContainer}>
                  {CATEGORIES_MAP[divisionCode] && (
                    <img
                      className={styles.img}
                      src={`/assets/img/categories/${CATEGORIES_MAP[divisionCode].imgName}.png`}
                      alt={name}
                    />
                  )}
                </div>
              </a>
            </Link>
          </div>
        ))}
      </Slider>
    </section>
  );
}
```

**Comment by nonoumasy** [comment]
> You have to use https://www.npmjs.com/package/@ant-design/react-slick to work with SSR without problems.
> 
> I was having problems with ssr-client prop matching while running the app on smaller devices
> 
> ```
> react-dom.development.js?61bb:88 Warning: Prop `src` did not match. Server: "/assets/img/categories/packaged.png" Client: "/assets/img/categories/baby.png"
> ```
> 
> Sample of source code
> 
> ```
> 
> const CATEGORIES_MAP = {
>   '01': {
>     imgName: 'vegetables',
>   },
>   '02': {
>     imgName: 'meat',
>   },
>   '03': {
>     imgName: 'milk',
>   },
>   '05': {
>     imgName: 'sausage',
>   },
>   '04': {
>     imgName: 'cheese',
>   },
>   '06': {
>     imgName: 'frozen',
>   },
>   '07': {
>     imgName: 'packaged',
>   },
> };
> 
> const SLIDER_SETTINGS = {
>   adaptiveHeight: true,
>   className: styles.slider,
>   infinite: true,
>   slidesToShow: 8,
>   slidesToScroll: 8,
>   speed: 500,
>   prevArrow: <PrevArrow category callback={() => arrowClick('categoryLeftClick')} />,
>   nextArrow: <NextArrow category callback={() => arrowClick('categoryRightClick')} />,
>   responsive: [
>     {
>       breakpoint: 1200,
>       settings: {
>         slidesToShow: 7,
>         slidesToScroll: 7,
>       },
>     },
>     {
>       breakpoint: 1120,
>       settings: {
>         slidesToShow: 6,
>         slidesToScroll: 6,
>       },
>     },
>     {
>       breakpoint: 1024,
>       settings: {
>         slidesToShow: 5,
>         slidesToScroll: 5,
>       },
>     },
>     {
>       breakpoint: 824,
>       settings: {
>         slidesToShow: 3,
>         slidesToScroll: 3,
>       },
>     },
>     {
>       breakpoint: 578,
>       settings: {
>         slidesToShow: 1,
>         slidesToScroll: 1,
>       },
>     },
>   ],
> };
> 
> export default function Categories({categories}) {
>   return (
>     <section className={styles.container}>
>       <Slider {...SLIDER_SETTINGS}>
>         {categories.map(({_id, divisionCode, name, numberOfProductsToDisplay, slug}) => (
>           <div key={`${_id}${numberOfProductsToDisplay}`}>
>             <Link href="/products/[...slug]" as={`/products/${slug}`}>
>               <a className={styles.tile}>
>                 <div className={styles.imgContainer}>
>                   {CATEGORIES_MAP[divisionCode] && (
>                     <img
>                       className={styles.img}
>                       src={`/assets/img/categories/${CATEGORIES_MAP[divisionCode].imgName}.png`}
>                       alt={name}
>                     />
>                   )}
>                 </div>
>               </a>
>             </Link>
>           </div>
>         ))}
>       </Slider>
>     </section>
>   );
> }
> ```

How is there 2 npm packages with the same name?

https://github.com/ant-design/react-slick
https://github.com/akiran/react-slick

**Comment by katsos** [comment]
@nonoumasy Nope. They are not identical my friend. I have solved all of my problems since I migrated to @ant-design/react-slick.

**Comment by zanedev** [comment]
+1 on @katsos recommendation, switched to `@ant-design/react-slick` and all my ssr issues went away magically in my gatsby app

**Comment by mandipredliodesigns** [comment]
nextjs 14.1.0 
react-slick 0.30.2
In development all things works proper
but in production it doesn't works proper 
![image](https://github.com/akiran/react-slick/assets/119833921/fe535406-5306-44a9-a170-c860b5abe146)
this is production build image
